### PR TITLE
[BugFix] fix cloud table does not support cross db colocation

### PR DESCRIPTION
**Pull request by starrocks-xupeng**
Fixes #issue

## What type of PR is this:
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Review by tracymacding — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/catalog/ColocateTableIndex.java [hunk 1]
```diff
@@ -244,12 +244,6 @@ public GroupId addTableToGroup(long dbId, OlapTable tbl, String groupName, Group
                 group2Schema.put(groupId, groupSchema);
             }
 
-            if (groupAlreadyExist) {
```
Comment (nshangyiming) [inline fe/fe-core/src/main/java/com/starrocks/catalog/ColocateTableIndex.java:247]: do we still need this check? assume we have a colocation group `groupa` which includes only olap tables, add we have a lake table `colocate_with='groupa'`, what will happen in this situation? is `assignedGroupId` null or not?

Comment (starrocks-xupeng) [reply fe/fe-core/src/main/java/com/starrocks/catalog/ColocateTableIndex.java:247]: this will not happen since we have run_mode in fe

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/ColocateTableIndex.java|2|2|100.00%|[]|

**Review by wyb — APPROVED** [review]

**Review by nshangyiming — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/ColocateTableIndex.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Pull request merged**